src/pages/transactions/[id].js

```diff
@@ -1,14 +1,24 @@
 import { useRouter } from "next/router";
+import { useState } from "react";
 import transactions from '../../data/transactions.json';
 import Link from "next/link";
 import ChevronLeft from "@/src/assets/icons/chevron_left.svg";
+import DeleteBin from "src/assets/icons/delete.svg"
 
-export default function TransactionDetails() {
+export default function TransactionDetails({handleDelete}) {
   const router = useRouter();
   const { id } = router.query;
+  const [modal, setModal] = useState(false);
+  const [deleteId, setDeleteId] = useState(null);
```

Comment:
I think you do not need to implement a status to handle the delete function and to pass the id of the deleted item as you you directly pass the id on click action. Check code suggestion below.

Reply:
No code suggestion found, but sounds good